src/dparse/parser.d

```diff
@@ -521,6 +521,11 @@ class Parser
         mixin(traceEnterAndExit!(__FUNCTION__));
         auto startIndex = index;
         AsmBrExp node = allocator.make!AsmBrExp();
+        if (!moreTokens)
```

Comment:
this should be done before the allocator call so we don't potentially leak memory